profiles/lib/relations.lua

```diff
@@ -0,0 +1,94 @@
+-- Profile functions dealing with various aspects of relation parsing
+--
+-- You can run a selection you find useful in your profile,
+-- or do you own processing if/when required.
+
+Utils = require('lib/utils')
+
+Relations = {}
+
+-- match ref values to relations data
+function Relations.match_to_ref(relations, ref)
+
+  function calculate_scores(refs, tag_value)
+    local tag_tokens = Set(Utils.tokenize_common(tag_value))
+    local result = {}
+    for i, r in ipairs(refs) do
+      local ref_tokens = Utils.tokenize_common(r)
+      local score = 0
+
+      for _, t in ipairs(ref_tokens) do
+        if tag_tokens[t] then
+          if Utils.is_number(t) then
+            score = score + 2
+          else
+            score = score + 1
```

Comment:
https://github.com/Project-OSRM/osrm-backend/pull/4478#discussion_r138778008

> In Project-OSRM/osrm-backend#4434, I proposed trying to match a prefix in the way’s `ref` to a component of the relation‘s `network` as an additional score to factor in. Via [this Overpass query](http://overpass-turbo.eu/s/rHB), I found [one way](http://www.openstreetmap.org/way/122502126) that’s simultaneously part of `PA 88 South` and `PA 88 Truck North`.
> 
> The state-by-state queries in https://github.com/Project-OSRM/osrm-text-instructions/issues/119#issuecomment-328053084 can only find relations tagged with directions in roles; they’d need to be reworked to handle `direction` tags on relations. But given that I’ve only found one problematic way in the U.S. so far, matching `networks` is probably overkill, even if it’d be more robust.

Reply:
So where i should to get `networks` tag?

Reply:
The [`network`](https://taginfo.openstreetmap.org/keys/network) tag primarily occurs on route relations, alongside the `direction` tag this PR already uses.

Reply:
Ok, i'll do that soon